src/LandIce/problems/LandIce_StokesFOBase.cpp

```diff
@@ -105,6 +105,17 @@ StokesFOBase (const Teuchos::RCP<Teuchos::ParameterList>& params_,
   // By default, we are not coupled to any other physics
   temperature_coupled = false;
   hydrology_coupled = false;
+
+  // Determine what Rigid Body Modes to compute and pass to the preconditioner
+  std::string rmb_list_name = "LandIce Rigid Body Modes For Preconditioner";
+  if(params->isSublist(rmb_list_name)) {
+    auto rbm_list = params_->sublist("LandIce Rigid Body Modes For Preconditioner");
+    computeConstantModes = params->sublist(rmb_list_name).get<bool>("Compute Constant Modes");
+    computeRotationModes = params->sublist(rmb_list_name).get<bool>("Compute Rotation Modes");
```

Comment:
Maybe set these to default to true and false, respectively just in case one is set in the sublist but not the other.

Reply:
I'd prefer this way. I think it is safer not to provide defaults  in the sublist, so that if one mispells the parameters, an error will be thrown. This could also be achieved by checking if the  parameters in the sublist are valid, but it would take a little more coding.

Reply:
Okay great, if the default is throw a readable error then it should be fine.